src/main/java/io/jenkins/plugins/intotorecorder/InTotoWrapper.java

```diff
@@ -416,8 +597,18 @@ public void tearDown(Run<?,?> build,
          */
         private String dumpLink(FilePath cwd) {
 
+            Gson gson = new GsonBuilder()
+            .serializeNulls()
+            // Use custom serializer to enforce non-floating point numbers
+            .registerTypeAdapter(Double.class, new NumericJSONSerializer())
+            .setPrettyPrinting()
+            .create();   
+
             try {
                 FilePath linkPath = cwd.child(link.getFullName());
+                FilePath provenancePath = cwd.child(this.stepName+".txt");
```

Comment:
The name should probably indicate it's SLSA provenance. And at least `.json` since provenance doesn't seem to use custom extensions.